ui/src/App.tsx

```diff
@@ -32,22 +32,28 @@ const App: React.FC = () => {
       };
 
       initPyodide();
-    } else {
-      try {
-        pyodide.current?.runPython(pythonSource);
+    }
+  }, [pyodideInitialized]);
+
+  const executePython = React.useCallback(() => {
```

Comment:
Don't get too confused by the terminology `"useCallback"`. This is just a way to create a memoized function, which only gets recomputed if one of the dependencies (`pythonSource` or `pythonError`) gets updated: https://reactjs.org/docs/hooks-reference.html#usecallback